### PR TITLE
Fix train dataloader

### PR DESCRIPTION
**Pull request by ntadimeti**
Fixing issue #207 

Print statement in scripts/worker.py after train_loader: 
```
    train_loader = torch.utils.data.DataLoader(
        train_dataset, batch_size=args.batch_size,
        shuffle=(train_sampler is None),
        num_workers=args.num_workers, pin_memory=True,
        sampler=train_sampler,
        drop_last=False)

    print (train_loader.dataset[0])
    print (train_loader.dataset[1])
```
Before fix output: 
```
{'idx': 0, 'input': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32), 'label_reg': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32), 'label_cla': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32)}
{'idx': 0, 'input': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32), 'label_reg': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32), 'label_cla': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32)}
```
After fix output:
```
{'idx': 0, 'input': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32), 'label_reg': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32), 'label_cla': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32)}
idx inside while loop:  1
file id:  0  local_idx:  1
{'idx': 1, 'input': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32), 'label_reg': array([0., 0., 0., ..., 1., 1., 1.], dtype=float32), 'label_cla': array([0., 0., 0., ..., 0., 0., 0.], dtype=float32)}
```
Notice the `idx` field being different and the `label_reg` field also being different.

### CODE REVIEW

File: tests/dl4atac/test_dataset.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                  "label_cla": np.array([1, 1, 0, 0])}
+    h5file = os.path.join(tmpdir, "h5file.h5")
+    h5io.dict_to_h5(input_dict, h5file)
+    train_dataset = DatasetTrain(files=[h5file], layers=None)
+    for idx in range(0, len(train_dataset)):
+        expected_dict = {"idx": idx,
+                         "input": input_dict["input"][idx],
+                         "label_reg": input_dict["label_reg"][idx],
+                         "label_cla": input_dict["label_cla"][idx]}
+        result_dict = train_dataset.__getitem__(idx)
```
Comment (tijyojwad) [inline tests/dl4atac/test_dataset.py:37]: `__getitem__` is just the implementation for `[]`. so `dataset.__getitem__(idx)` is the same as `dataset[idx]`

File: tests/dl4atac/test_dataset.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    input_dict = {"input": np.array([0, 1, 3, 4]),
+                  "label_reg": np.array([1, 2, 10, 1]),
+                  "label_cla": np.array([1, 1, 0, 0])}
+    h5file = os.path.join(tmpdir, "h5file.h5")
+    h5io.dict_to_h5(input_dict, h5file)
+    infer_dataset = DatasetInfer(files=[h5file], layers=None)
+    for idx in range(0, len(infer_dataset)):
+        expected_dict = {"idx": idx,
+                         "input": input_dict["input"][idx]}
+        result_dict = infer_dataset.__getitem__(idx)
```
Comment (tijyojwad) [inline tests/dl4atac/test_dataset.py:57]: same here, replace `__getitem__(idx)` with `[idx]`

File: tests/dl4atac/test_dataset.py (hunk 1 above)
Comment (ntadimeti) [reply tests/dl4atac/test_dataset.py]: Oh, right. changed.

File: tests/dl4atac/test_dataset.py (hunk 2 above)
Comment (ntadimeti) [reply tests/dl4atac/test_dataset.py]: oh right, changed.

**Pull request merged**